cmd/syncthing/main.go

```diff
@@ -700,19 +700,26 @@ func syncthingMain() {
 		connectionSvc.Add(relaySvc)
 	}
 
-	// Start discovery
+	// The externalAddr tracks our external addresses for discovery purposes.
 
-	localPort := addr.Port
-	discoverer = discovery(localPort, relaySvc)
+	extAddr := new(externalAddr)
```

Comment:
Yeah... We should have a constructor. I did it this way because we only want to set the upnpSvc if we're actually using UPnP. With the constructor, we'll have to wrap the constructor call in the `if upnpEnabled` thing, but I'll look at it.